react/BottomSheet/BottomSheet.jsx

```diff
@@ -72,7 +72,8 @@ const createStyles = ({ squared, hasToolbarProps }) => ({
     backgroundColor: 'var(--secondaryTextColor)'
   },
   stack: {
-    backgroundColor: 'var(--defaultBackgroundColor)'
+    backgroundColor: 'var(--defaultBackgroundColor)',
+    paddingBottom: immersive ? 'var(--flagship-bottom-height)' : 0
```

Comment:
I don't know why do we need to test this `immersive` stuff. Don't the `--flagship-bottom-height` is updated by the flagship app by itself? If flagship detect that this is not in immersive mode then the value is set to 0? 

@acezard @Ldoppea

Reply:
> No, the --flagship-bottom-height is instantiated at boot and never updated. 

Are you sure about that? For what I can read, we inject this CSS variable when serving it from the http server: https://github.com/cozy/cozy-flagship-app/blob/master/src/libs/httpserver/httpServerProvider.js#L107. 

This method is called from the `CozyProxyWebView` and each time the slug changes, this method is called. So the behavior is dynamic each time the user moves from an app to an other, no? If yes, then we can rely on it. 

cc @Ldoppea

Reply:
Yes from that point of view you're right, but I was referring to the raw pixel value: https://github.com/cozy/cozy-flagship-app/blob/master/src/libs/dimensions.js#L16

It is created once and does not support change during runtime

Reply:
> It is created once and does not support change during runtime

Yep, but we don't need it. Since it won't change, right?

Reply:
> > It is created once and does not support change during runtime
> 
> Yep, but we don't need it. Since it won't change, right?

Yes, so the value will always be the same number in immersive or not immersive webview

Reply:
> Yes, so the value will always be the same number in immersive or not immersive webview.

I don't get it. 

In immersive, the value will be the result of `getNavbarHeight()`. So it can be 0 or an other value depending on the device. 

In not immersive webview, the variable will not be defined, so we'll have a 0 value. 

Is that right?

Reply:
Yes, that is correct because we inject the variable only on `home` slug. So technically we could ignore the immersive or not and only fetch the CSS value. It would currently work but I think it's more clear if we check for immersive presence. In the other case, if we just fetch the CSS value (that exists or does not exists), we give the impression we always want that value, not knowing it does not exist in certain contexts.
But all this is up to opinion

Reply:
> not knowing it does not exist in certain contexts.

Maybe this is what we should fix. Always having this var defined, but its value can change for each app. Like that, we always use it, we don't need to ask ourselves if we're immersive or not. The component should handle this value as it should handle the `safe-area` stuff.

Reply:
> > not knowing it does not exist in certain contexts.
> 
> Maybe this is what we should fix. Always having this var defined, but its value can change for each app. Like that, we always use it, we don't need to ask ourselves if we're immersive or not. The component should handle this value as it should handle the `safe-area` stuff.

ok, it would be a simple fix in the native app to add a 0 value to this CSS var if we don't need it.
@zatteo in that case you can remove mention to immersive and just use the CSS var

Reply:
I agree that this immersive variable is a bit redundant. It give more context, but maybe too much? Do we really want to know from apps that the margin should be applied for "immersive" purpose? Or do we just need to know that we can manipulate a value that ensure "safe area"?